meson.build

```diff
@@ -26,6 +26,13 @@ if host_system == 'linux'
   _args += '-Wno-unused-const-variable'
 endif
 
+if cppc.get_id() == 'msvc'
+  add_project_arguments(cppc.get_supported_arguments('/W4'), language : 'cpp')
+  add_project_arguments('/wd4018', '/wd4061', '/wd4100', '/wd4101', '/wd4127', '/wd4189', '/wd4244', '/wd4245', '/wd4267', '/wd4324', '/wd4365', '/wd4389', '/wd4456', '/wd4457', '/wd4458', '/wd4459', '/wd4514', '/wd4701', '/wd4820', language: 'cpp')
+  add_project_arguments('/MP', language: 'cpp')
```

Comment:
The use of `/MP` seems wrong here. Isn't this supposed to instruct cl.exe to spawn multiple copies and compile each source file with one process, when given multiple source files on the command line?

Meson's ninja backend already creates one build edge per source file, it doesn't need a command-line option to cl.exe to unlock parallel compilation since ninja can already do this quite well.

Meson's Visual Studio backend uses whatever parallelism msbuild makes available, which is less... robust. It would benefit from /MP except that it doesn't actually work -- meson specifies output filenames for compiled object files, which surprisingly does mean that /MP gets ignored (!) so it does nothing. Meson does automatically set up a slightly newer approach to parallelized builds, though, see https://github.com/mesonbuild/meson/commit/14096260c434a6fcc80d06eeba006bd1965b213c. It apparently requires  Visual Studio 2019 16.3, see e.g. https://devblogs.microsoft.com/cppblog/improved-parallelism-in-msbuild/

tl;dr setting /MP does nothing but make command lines a bit bigger. I would suggest dropping this argument. For ninja builds, as well as for sufficiently *new* versions of VS, build parallelism works regardless, and you cannot really fix old versions of VS.